Revamp blog page look & feel.

geeknoid opened the pull request:
- The blog page now has a sidebar showing all blog posts.

- Blog entries now have a clean view of publication date, attribution, and subtitle

- Sidebar entries, both on the blog page and in the docs area now show tooltips when hovering the
mouse over the entry.

Staging site: https://geeknoid.github.com/istio.github.io

PS: Please give the staging site a good run-through so make sure I didn't mess up any edge cases.

ldemailly commented on _data/istio.yml:
```diff
@@ -1 +1,2 @@
 version: 0.3.0
+latest_blog_post: /blog/posts/2017/mixer-spof-myth.html
```
add the missing newline?

geeknoid replied:
Whoops, that was a failed experiment, so I just reverted the change completely.

ldemailly commented on _docs/concepts/index.md:
```diff
@@ -10,4 +10,5 @@ type: markdown
 
 ---
 
+{% assign docs = site.docs %}
```
what does this do / why is it needed ? what happens if you missed one ?

geeknoid replied:
Turns the sidebar functionality into a function. Whereas before, the sidebar was closely coupled with our "docs" tree, it's now more general-purpose and can be used with any hierarchy of documents (such as blogs).

If you omit this statement, you'll end up with a blank sidebar.

rshriram commented:
missing image in https://geeknoid.github.io/istio.github.io/blog/posts/2017/0.1-using-network-policy.html

rshriram commented on _blog/posts/2017/0.1-auth.md:
```diff
@@ -1,12 +1,17 @@
 ---
-title: "Istio Auth: secure by default service to service communications"
+title: "Istio Auth"
```
rename to Using Istio to improve end-to-end Security ?

rshriram commented on _blog/posts/2017/0.1-announcement.md:
```diff
@@ -1,12 +1,17 @@
-layout: post
+publish_date: May 24, 2017
+subtitle: A robust service mesh for microservices
+attribution:
```
Martin Taill.., Louis Ryan, Shriram Rajagopalan. ??

rshriram replied:
Possibly X, Company format? Or should we dump the company?

ldemailly replied:
I think "The Istio Team" is more appropriate than 3 individuals for the 0.1 announcement

geeknoid replied:
I went with "The Istio Team" for the release announcements.

rshriram commented on _blog/posts/2017/0.1-canary.md:
```diff
@@ -1,11 +1,15 @@
-layout: post
+publish_date: June 14, 2017
+subtitle:
+attribution:
```
Frank Budinsky

rshriram replied:
Actually, Frank Budinsky, IBM

rshriram commented on _blog/posts/2017/0.1-using-network-policy.md:
```diff
@@ -1,11 +1,15 @@
-layout: post
+publish_date: August 10, 2017
+subtitle:
+attribution:
```
Spike Curtis

geeknoid replied:
Done

ldemailly commented:
if you add the missing trailing / to https://www.weave.works/blog/istio-weave-cloud/ it should resolve the circleci failure

but I wonder how come it doesn't flag the missing image / is it some JS instead of plain html ?

geeknoid commented:
@ldemailly The link tracking appears to have some issues. You mentioned here to add a trailing / to fix the weavecloud link. I added the link as requested and the link works fine. The issue is that I hadn't touched that text at all, I just moved the blog post to a different location.

And as you point out, it didn't flag the missing image. It's conceivable that the link checker is only looking at <a href> and not <img src>.
